### PR TITLE
Refactor to add Mutant::Mutator.mutate(...)

### PR DESCRIPTION
**Pull request by backus**
This PR removes the `Mutant::Mutator::Registry#call` method in favor of `Mutant::Mutator.mutate`. We discussed this here: https://github.com/mbj/mutant/pull/565#discussion-diff-61762942

### CODE REVIEW

File: spec/unit/mutant/registry_spec.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    it 'returns registered mutator' do
+      expect(lookup).to be(mutator)
+    end
+
+    it_behaves_like 'a command method'
+
+    context 'when registered twice' do
+      it 'fails upon registration' do
+        expect { register_mutator }.to raise_error(described_class::RegistryError, 'Duplicate type registration: :true')
```
Comment (mbj) [inline spec/unit/mutant/registry_spec.rb]: I know that rubocop on mutant right now does not enforce it, but could we change these overlong (and the similar ones below) lines to conform to 79 chars?

File: spec/unit/mutant/subject_spec.rb [hunk 2]
```diff
@@ -71,7 +71,7 @@ def foo
     subject { object.mutations }
 
     before do
-      expect(Mutant::Mutator::REGISTRY).to receive(:call).with(node).and_return([mutation_a, mutation_b])
+      expect(Mutant::Mutator).to receive(:mutate).with(node).and_return([mutation_a, mutation_b])
```
Comment (mbj) [inline spec/unit/mutant/subject_spec.rb]: Can we reformat this to conform to 79 chars?

File: lib/mutant/registry.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    RegistryError = Class.new(TypeError)
+
+    # Register class for AST node class
+    #
+    # @param [Symbol] type
+    # @param [Class] class
+    #
+    # @return [self]
+    def register(type, klass)
+      fail RegistryError, "Invalid type registration: #{type.inspect}" unless AST::Types::ALL.include?(type)
```
Comment (mbj) [inline lib/mutant/registry.rb]: I wounder if these can be reformatted to 79 chars easily?

Comment (backus) [inline lib/mutant/registry.rb]: Sure I can do that. I thought this project preferred 120 chars but I can make the change

**Pull request merged**